modules/libcom/src/osi/os/posix/osdThread.c

```diff
@@ -667,12 +726,28 @@ epicsShareFunc epicsThreadBooleanStatus epicsShareAPI epicsThreadHighestPriority
 {
     unsigned newPriority = priority - 1;
 #if defined(_POSIX_THREAD_PRIORITY_SCHEDULING) && _POSIX_THREAD_PRIORITY_SCHEDULING > 0
-    int diff;
-    diff = pcommonAttr->maxPriority - pcommonAttr->minPriority;
-    if(diff<0) diff = -diff;
-    if(diff>1 && diff <100) newPriority -=  100/(diff+1);
+    if ( pcommonAttr->valid ) {
+        int newOss;
+        if ( epicsThreadPriorityMax - epicsThreadPriorityMin > pcommonAttr->maxPriority - pcommonAttr->minPriority ) {
```

Comment:
I'm confused by this condition.  Could this be simplified by working with `priority` as a real number here, as well as in `osi2posixPriority()` where rounding is already happening?

Reply:
The way I understand epicsThreadLowestPriorityLevelAbove/Below() is that it should find the next priority level that is possible on the particular implementation. Rounding must be avoided (otherwise the result may be the same priority). The intention here is to take a one-sized step in the coarser domain (osi vs. oss). I.e., if there are  more OSI priority levels than OSS levels then we map the current OSI -> OSS, take one step in the OSS domain and map back. If OSS is finer grained than OSI then we can just take one OSI step.

A brute-force version could be

````
currentOSS = osi2PosixPriority( priority );
 do {
  if ( ++priority > epicsThreadPriorityMax)
    return epicsThreadBooleanStatusFail;
} while ( osi2PosixPriority( priority ) == currentOss );
  
 *pPriority = priority;
return epicsThreadBooleanStatusSuccess;
````

Reply:
This block definitely needs a comment stating the "coarser domain" intent.  It makes sense to me when you describe it in those terms.

Also, seeing the added complexity in common between these two functions.  Would it be better to factor out this common code as a single private helper function into which is passed either +1 or -1 ?

Reply:
I'll look into that.

Reply:
Done